src/event-log/subscription.ts

```diff
@@ -0,0 +1,84 @@
+import type { DwnError } from '../core/dwn-error.js';
+import type { EventEmitter } from 'events';
+import type { MessageStore } from '../types/message-store.js';
+import type { EmitFunction, SubscriptionHandler } from '../types/subscriptions.js';
+import type { Filter, KeyValues } from '../types/query-types.js';
+import type { GenericMessage, GenericMessageHandler } from '../types/message-types.js';
+
+import { FilterUtility } from '../utils/filter.js';
+
+/**
+ * Base class to extend default subscription behavior.
+ */
+export class SubscriptionHandlerBase implements SubscriptionHandler {
+  protected eventEmitter: EventEmitter;
+  protected messageStore: MessageStore;
+  protected filters: Filter[];
+  protected tenant: string;
+  protected message: GenericMessage;
+
+  #unsubscribe: () => Promise<void>;
+  #id: string;
```

Comment:
The use of # syntax was once undone when I started using it (https://github.com/TBD54566975/dwn-sdk-js/pull/466), while requirements may have changed, I'd use the _ prefix instead even if just for the sake of consistency. Also, support for it in VSC debugging isn't quite seamless yet.